entities/src/main/java/com/flipkart/varadhi/entities/utils/HeaderUtils.java

```diff
@@ -0,0 +1,89 @@
+package com.flipkart.varadhi.entities.utils;
+
+import com.flipkart.varadhi.entities.config.MessageConfiguration;
+import com.flipkart.varadhi.entities.constants.MessageHeaders;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+
+import java.util.List;
+import java.util.Map;
+import lombok.Getter;
+
+public class HeaderUtils {
```

Comment:
lets move it out of Util and name it appropriately.
This should be called HeadersCtx